crates/matrix-sdk-crypto/src/types/events/room_key.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types for `m.room_key` to-device events.
+
+use std::collections::BTreeMap;
+
+use ruma::{serde::Raw, EventEncryptionAlgorithm, OwnedRoomId, RoomId};
+use serde::{Deserialize, Serialize};
+use serde_json::{value::to_raw_value, Value};
+use vodozemac::megolm::SessionKey;
+
+use super::{EventType, ToDeviceEvent};
+
+/// The `m.room_key` to-device event.
+pub type RoomKeyEvent = ToDeviceEvent<RoomKeyContent>;
+
+impl RoomKeyContent {
```

Comment:
It's pretty uncommon for `impl Foo` blocks to come before `Foo`s definition, is there a reason for doing it this way here?

Reply:
Fixed in https://github.com/matrix-org/matrix-rust-sdk/pull/767/commits/3e98583b1559386ba6a5be82b9b6e215bda4c00a